src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,556 @@
+
+    /** @var array A list of allowed frequencies used to configure custom expressions */
+    protected $customFrequencies = [];
+
+    /** @var array */
+    protected $advanced = [];
+
+    /** @var array */
+    protected $regulars = [];
+
+    /** @var string Schedule frequency of this element */
+    protected $frequency = self::NO_REPEAT;
+
+    /** @var string */
+    protected $customFrequency = RRule::DAILY;
+
+    /** @var DateTime */
+    protected $start;
+
+    /** @var DateTime */
+    protected $end;
+
+    /** @var WeeklyFields Weekly parts of this schedule element */
+    protected $weeklyField;
+
+    /** @var MonthlyFields Monthly parts of this schedule element */
+    protected $monthlyFields;
+
+    /** @var AnnuallyFields Annually parts of this schedule element */
+    protected $annuallyFields;
+
+    protected function init(): void
+    {
+        $this->start = new DateTime();
+        $this->weeklyField = new WeeklyFields('weekly-fields', [
+            'default'   => $this->start->format('D'),
+            'protector' => function (string $day) {
+                return $this->protectId($day);
+            },
+        ]);
+
+        $this->monthlyFields = new MonthlyFields('monthly-fields', [
+            'default'         => $this->start->format('j'),
+            'availableFields' => (int) $this->start->format('t'),
+            'protector'       => function ($day) {
+                return $this->protectId($day);
+            }
+        ]);
+
+        $this->annuallyFields = new AnnuallyFields('annually-fields', [
+            'default'   => $this->start->format('M'),
+            'protector' => function ($month) {
+                return $this->protectId($month);
+            }
+        ]);
+
+
+        $this->regulars = [
+            RRule::MINUTELY  => $this->translate('Minutely'),
+            RRule::HOURLY    => $this->translate('Hourly'),
+            RRule::DAILY     => $this->translate('Daily'),
+            RRule::WEEKLY    => $this->translate('Weekly'),
+            RRule::MONTHLY   => $this->translate('Monthly'),
+            RRule::QUARTERLY => $this->translate('Quarterly'),
+            RRule::YEARLY    => $this->translate('Annually'),
+        ];
+
+        $this->customFrequencies = array_slice($this->regulars, 2);
+        unset($this->customFrequencies[RRule::QUARTERLY]);
+
+        $this->advanced = [
+            static::CUSTOM_EXPR => $this->translate('Custom…'),
+            static::CRON_EXPR   => $this->translate('Cron Expression…')
+        ];
+    }
+
+    public function setDefaultElementDecorator($decorator)
+    {
+        parent::setDefaultElementDecorator($decorator);
+
+        $this->weeklyField->setDefaultElementDecorator($this->getDefaultElementDecorator());
+        $this->monthlyFields->setDefaultElementDecorator($this->getDefaultElementDecorator());
+        $this->annuallyFields->setDefaultElementDecorator($this->getDefaultElementDecorator());
```

Comment:
This isn't necessary. The default element decorator is automatically propagated to fieldsets.